### PR TITLE
Apply loadmore and load by genres for search screen, save search histories

### PR DESCRIPTION
**Pull request by Vanh1200**
## [Pull request title]
Apply loadmore and load by genres for search screen
### [Redmine link]
https://edu-redmine.framgia.vn/issues/7975

### CODE REVIEW

**Comment by Vanh1200** [comment]
ready

File: app/src/main/java/com/ptit/filmdictionary/data/source/local/DataBaseHelper.java [hunk 1]
```diff
@@ -118,4 +129,51 @@ public boolean isFavorite(Movie movie) {
                 null);
         return (cursor != null && !cursor.isAfterLast());
     }
+
+
```
Comment (NguyenMinhTuanB) [inline app/src/main/java/com/ptit/filmdictionary/data/source/local/DataBaseHelper.java]: remove bl

File: app/src/main/java/com/ptit/filmdictionary/data/source/local/DataBaseHelper.java [hunk 2]
```diff
@@ -118,4 +129,51 @@ public boolean isFavorite(Movie movie) {
...
+                    .getColumnIndexOrThrow(HistoryEntity.TITLE));
+            History history = new History(id, title);
+            histories.add(history);
+            cursor.moveToNext();
+        }
+        historiesObservable.addAll(histories);
+        return historiesObservable;
+    }
+
+    public boolean addHistory(History history) {
```
Comment (NguyenMinhTuanB) [inline app/src/main/java/com/ptit/filmdictionary/data/source/local/DataBaseHelper.java]:
```suggestion
    public boolean addedHistory(History history) { or  public boolean hasAddedHistory(History history) {
```

File: app/src/main/java/com/ptit/filmdictionary/data/source/local/DataBaseHelper.java [hunk 3]
```diff
@@ -118,4 +129,51 @@ public boolean isFavorite(Movie movie) {
...
+        SQLiteDatabase db = getWritableDatabase();
+        ContentValues values = new ContentValues();
+        values.put(HistoryEntity.TITLE, history.getTitle());
+        long result = db.insert(HistoryEntity.TABLE_HISTORY, null, values);
+        return (result > 0);
+    }
+
+    public boolean deleteHistory(History history) {
```
Comment (NguyenMinhTuanB) [inline app/src/main/java/com/ptit/filmdictionary/data/source/local/DataBaseHelper.java]: tuương tự

**Comment by Vanh1200** [comment]
ready

**Review by NguyenMinhTuanB — APPROVED** [review]

**Pull request merged**